frontend/tsconfig.json

```diff
@@ -44,8 +44,10 @@
 
     /* Module Resolution Options */
     "moduleResolution": "node",                     /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
-    "baseUrl": "src/",                              /* Base directory to resolve non-absolute module names. */
-    // "paths": {},                                 /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    // "baseUrl": "src/",                           /* Base directory to resolve non-absolute module names. */
+    "paths": {                                      /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+      "~/*": ["./src/*"]
+    },
```

Comment:
I can't say that I'm fond of this change, but that's just an opinion.

The whole tilde import stuff .. I just find it confusing.

In another project, there's actual names to this, like, `_core/*` and `_modules/*`, though I'm not sure this is really needed.

Is this just to avoid name clashes between nodejs packages and our source code?

Reply:
Yeah, pretty much. Tbh, the actual fix I'd prefer would be to de-complicate the structure enough to not need anything except for relative paths.

Very much open to discussing alternatives on this, or to reverting this change.

Reply:
Not a fan either. If this can be reverted without causing clashes, I'd be in favor of that.

Reply:
One small DX issue with using undecorated imports for the root-level directories is that automated tooling won't be able to differentiate between them, and so e.g. will think that `core/utils` ought to come before `react`. For example, in VS Code if the imports are in order, then its tooling will also add new imports to appropriate places from autocompletion, and `~` is after the letters in Latin-1 (though `_` appears to be special-cased to work the same way). Otherwise imports are added to the end of the list, and need to be manually moved around.

So while yes, we can keep with the old pattern, it does mean mixing the app's internal and dependency namespaces, and it means that tooling won't work quite as well.

Reply:
That's sounds like a reasonable argument to me. Let's go ahead, but please add a comment about the imports. Maybe in the CONTRIBUTING file.